src/app/consumption-estimation/consumption-estimation.component.ts

```diff
@@ -35,32 +27,22 @@ export class ConsumptionEstimationComponent implements OnInit {
   customerData: Customer[]=[] ;  
   fullAddresses: Address[]=[];
 
-  selectedAddressID:number=-1;
+  selectedAddressID:number=0;
   selectedMeterNr:number=1;
   answerRadio:number=0;
   loggedInUser!: User;
 
-  //received from the post req
-  contract_id:number=0;
-
-  meters: Meter[]=[];
-  contract: EstimatedContract=
+  estimationRegistration:EstimationRegistration =
   {
-      start_date: new Date(),
-      end_date: new Date(),
-      customer_type: '',
-      tariff_id: -1 ,
-      estimation_id:-1,
-      address_id:-1,
-      service_type:-1,
-      status:"inactive",
-      building_type: -1,
-      family_size: -1 ,
-      equipments: '',
-      past_consumption: -1,
-      estimated_consumption:-1
-  
+    service_type : 0, 
+    building_type : 0, 
+    family_size: 0, 
+    equipment: [],
+    past_consumption: 0, 
+    meters: []
   };
+  
+  meters: Meter[]=[];
```

Comment:
Use local variables in their functions where they are supposed to be instead of "global".
Due to time contraints u can skip this